src/transforms/api/_transform.py

```diff
@@ -218,10 +273,95 @@ def filesystem(self):
             FileSystem:
                 A `FileSystem` object for reading from `Foundry`.
         """
-        return FileSystem(
-            base_path=get_dataset_path(
-                foundry_dev_tools.config.Configuration["cache_dir"],
-                self._dataset_identity,
+        return FileSystem(base_path=self._local_path)
+
+
+class LightweightTransformInput(TransformInput):
+    """LightweightTransformInput class, passed when using @lightweight decorator."""
+
+    def __init__(self, input_arg: Input):
+        super().__init__(input_arg)
+        self.branch = input_arg.branch
+        self.rid = input_arg.get_dataset_identity()["dataset_rid"]
+        del self.path  # we change the field to a property in this class
+
+    @cached_property
+    def _parquet_files(self) -> List[Path]:
+        return list(Path(self._local_path).glob("**/*.parquet"))
+
+    @cached_property
+    def _csv_files(self) -> List[Path]:
+        return list(Path(self._local_path).glob("**/*.csv"))
+
+    def dataframe(self):
+        raise NotImplementedError(
+            "Lightweight transforms do not support the dataframe method."
+        )
+
+    def path(self) -> str:
```

Comment:
Is `LightweightTransformInput` changing the API spec of `TransformInput` where `path` is defined as "The Project path of the input dataset."
If yes, we can keep the code as it is here.

https://www.palantir.com/docs/foundry/transforms-python/transforms-python-api-classes/#transforminput

Reply:
Yes, we should update the public docs to include the APIs of `LightweightTransformInput` and output, I'll add it to our todo list

Reply:
Just wanted to edit that I now saw the new public docs about input.path() here:

https://www.palantir.com/docs/foundry/transforms-python/lightweight-examples/#working-with-apache-datafusion-and-duckdb

In this case the code here is fine. (Although I don't like that you re-purposed the `path` property ;) )